lib/carconnectivity.h

```diff
@@ -52,13 +77,79 @@ class CarConnectivity : public GridConnectivity< Eigen::Vector3d, Eigen::Matrix3
    * making the motion primitives
    * @param onlyfwd If true, then only +ve forward velocity is used
    */
-  CarConnectivity(const CarGrid& grid,
+  CarConnectivity(const CarGrid& grid,const CarCost& cost,
                   double dt = .25,
                   double vx = 5,
                   double kmax = 0.577,
                   int kseg = 4,
-                  bool onlyfwd = false);
-  
+                  bool onlyfwd = false,
+                  bool allow_slip = true)
+  : grid(grid),cost(cost), allow_slip(allow_slip) {
+    SetPrimitives(dt, vx, kmax, kseg, onlyfwd);
+  }
+
+
+  /**
+   * Generates primitives based on the configuration
+   * @param dt time duration
+   * @param cfg the configuration for generation of primitives
+   * @return
+   */
+  bool SetPrimitives(double dt, CarPrimitiveConfig& cfg){
```

Comment:
I have different connection types and different data type, for which most of the code is identical.  What would be the best way to implement is without code duplication?